SportsHub/SportsHub.Application/Services/ArticleService.cs

```diff
@@ -16,5 +16,13 @@ public ArticleService(IUnitOfWork unitOfWork)
         {
             return await _unitOfWork.ArticleRepository.GetByTitleAsync(title);
         }
+
+        public async Task<bool> DeleteArticleAsync(int id)
+        {
+            var articleExists = await _unitOfWork.ArticleRepository.DeleteArticleAsync(id) is not null;
+            if (!articleExists) return false;
+            await _unitOfWork.SaveChangesAsync();
```

Comment:
how does it suppose to work?
in yours DeleteArticleAsync - you return value. then just SaveAsync.
1. add getting method
2. if no null, then add delete action to the repo -> then calls Save async

Reply:
and why do we need a Generic repository at all? please makes methods there Async and use them/ The is GetById and Remove actions

Reply:
Okey i will do it

Reply:
I did not changed the methods in GenericRepository but i used FirstOrDefaultAsync to find the article then used the Delete method from GenericRepository in my Delete method in ArticleRepository which is changed to return bool too . Then in service i call the method from the repo and check if is true or false then use the UOF to save changes like to save the database after we delete the article . Later we can throw exception in the Service instead of repeating to return false there